angular-intro-project/src/app/courses-list/courses-list.component.ts

```diff
@@ -10,14 +11,20 @@ export class CoursesListComponent implements OnInit {
 
   readonly DESCRIPTION = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur'
   items: Array<Course>;
+  inputText: string;
 
-  constructor() { }
+  constructor(private inputService: InputService) { 
+    this.inputService.input$.subscribe((data) => {
```

Comment:
Please, move this logic into OnInit lifecycle hook. We mostly use constructor in component for DI.